src/basic/parse-util.c

```diff
@@ -807,3 +809,32 @@ bool nft_identifier_valid(const char *id) {
                         return false;
         return true;
 }
+
+bool machine_spec_valid(const char *spec) {
+        const char *h;
+
+        /* Checks if a container specification in the form "user@container" or just "container" is valid.
+         *
+         * If the "@" syntax is used we'll allow either the "user" or the "container" part to be omitted,
+         * but not both. */
+
+        assert(spec);
+
+        h = strchr(spec, '@');
+        if (!h)
+                h = spec;
+        else {
+                const char *user = strndupa_safe(spec, h - spec);
```

Comment:
hmm, this is directly user controlled data, let's not use stack memory for this please.